### PR TITLE
URI

### PR DESCRIPTION
**Pull request by mwmiller**
- Improve version reasoning
- Improve URI parsing

### CODE REVIEW

**Comment by mwmiller** [comment]
Any updates on this?

File: t/cache.t [hunk 1]
```diff
@@ -79,7 +77,6 @@ isnt $new_cache, $cache, "Got new cache object";
 
 ok(Cache::RedisDB->set("Test", "TTL", "I will expire", 60), "Set TTL test key for 60 second expiration.");
 is(Cache::RedisDB->get("Test", "key1"), "value1", "Got value1 for Test::key1");
-is(Cache::RedisDB->ttl("Test", "TTL"), 59, "A moment later the expiration of TTL is down to 59 seconds.");
```
Comment (tfoertsch) [inline t/cache.t]: Maybe this test should be something like

```perl
cmp_ok Cache::RedisDB->ttl("Test", "TTL"), '<=', 60, "Read back TTL";
```

Not sure if removing it for good is a good idea.

**Comment by tfoertsch** [comment]
@mwmiller except for an error message it seems completely compatible to the old version. Why did you remove the `TTL` test?

File: lib/Cache/RedisDB.pm [hunk 2]
```diff
@@ -149,7 +153,8 @@ Return a list of all known keys in the provided I<$namespace>.
 sub keys {
     my ($self, $namespace) = @_;
     my $prefix = _cache_key($namespace, undef);
-    return [map { s/^$prefix//; $_ } @{redis->keys($prefix . '*')}];
+    my $pl = length($prefix);
+    return [map { substr($_, $pl) } @{redis->keys($prefix . '*')}];
```
Comment (mwmiller) [inline lib/Cache/RedisDB.pm]: Yes.  Large key lists (we have several hundred thousand in some namespaces) see a real difference with this change.

File: t/cache.t (hunk 1 above)
Comment (mwmiller) [inline t/cache.t]: I originally switched it to `60`, but the test removed here already exists: see L115.

**Pull request merged**